### PR TITLE
[nic.pl] Strip whitespace in .nicrc keys and values

### PR DESCRIPTION
**Pull request by ChristopherA8**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This fixes an issue where values are ignored if a user accidentally adds a whitespace at the beginning of a line in .nanorc

Where has this been tested?
---------------------------
**Operating System:** macOS Sonoma

### CODE REVIEW

File: bin/nic.pl
Comment (L1ghtmann) [inline bin/nic.pl:319]: We can avoid having to do this by also matching anything with a leading space, as you handle the leading/trailing below
```perl
/^\s*(.+?)\s*=\s*"(.*)"$/
```

File: bin/nic.pl
Comment (L1ghtmann) [inline bin/nic.pl:324]: With the above change, this would no longer be necessary as the key is matched irrespective of whitespace and the equal sign is also cleared of whitespace before splitting the pair

File: bin/nic.pl
Comment (L1ghtmann) [inline bin/nic.pl:325]: This seems like a worthwhile change, but I'm not sure if format-correcting the user configuration at load is something we want to be responsible for or not. Will leave it up to an additional set of eyes, but `dm.pl` should catch anything that's wrong with the control config and the other options are boolean and appear to work as expected irrespective of whitespace.

**Review by L1ghtmann — CHANGES_REQUESTED** [review]

**Comment by L1ghtmann** [comment]
Hey, thanks for catching this! Had a handful of minor comments. Will likely want an additional review before merging.

File: bin/nic.pl
Comment (L1ghtmann) [reply bin/nic.pl:325]: Re other options: https://theos.dev/docs/nic#nicrc

**Comment by L1ghtmann** [comment]
@ChristopherA8 would you mind taking a look at the attached comments when you have a moment. Thanks again!

**Review by leptos-null — APPROVED** [review]
Looks good to me.

Tested with

```
username = "Leptos <leptos.0.null@gmail.com>"
```

```
    username = "Leptos <leptos.0.null@gmail.com>"
```

```
# username = "Leptos <leptos.0.null@gmail.com>"
```

**Comment by L1ghtmann** [comment]
Merged with a couple of modifications. Thanks again!

**Pull request merged**
